extendr-api/src/wrapper/nullable.rs

```diff
@@ -77,3 +131,88 @@ where
         }
     }
 }
+
+impl<T> Into<Option<T>> for Nullable<T> {
```

Comment:
I'm worried that we're using `Optional<T>` to represent a `T` that may be `NA` elsewhere in extendr. Should we consider adding a counterpart to this `Nullable<T>` that's like `NAable<T>` so we can drop the general `Option` conversion?

Reply:
A valid point. However, due to the lack of docs, people just prefer to use known types, like `Option<i32>` not `Rint`. I am honestly OK with dropping options conversion at all, but it would be nice to have more methods to work with nullable type, like `map`, `and_then`, etc.

Reply:
I think conversion from `Nullable` to `Option` is fine and idiomatic Rust. I just think we shouldn't have an automatic conversion from `Robj` to `Option<T>` specifically. Rather we should have an intermediate `NAable`. But this is probably a discussion for elsewhere.

Reply:
Actually on second thoughts, it's not as confusing as it seems. When we have a vector, a given element can be `NA`, but a given element can't be `NULL`. So there's never a point at which there is confusion between `NA` and `NULL`. 

For example when we convert a `Robj` to `Optional<Vector<i32>>` the "optional" part is very clearly "is this null or not", because being `NULL` means there is no vector at all. If we convert `Robj` to `Vector<Optional<i32>>` the optional part is "is this particular element `NA`". So as long as this is documented, I think it's fine.

Do you agree?

Reply:
Should we disallow conversion from `Option` to `Nullable` then? In that case, `Option` might represent `NA`-aware value.

Reply:
No, I don't think so. There shouldn't ever be overlap between `Option<T>` derived from `Nullable` and `Option<T>` derived from `NA`, so it's not confusing.